File: Moncho_MATLAB/R7.m
function im_filtro = R7(im_rule_7)

[fil, col, ch] = size(im_rule_7);
im_YCbCr = rgb2ycbcr(im_rule_7);
im_filtro = double(zeros(fil, col));

for x = 1:fil
    for y = 1:col
        if im_YCbCr(x, y, 2) <= 120  &&  im_YCbCr(x, y, 3) >= 150
            im_filtro(x, y) = 1;
        else
            im_filtro(x, y) = 0;
        end
    end
end

imshow(im_filtro)
end

File: Moncho_MATLAB/R7.py
import numpy as np
import matplotlib.pyplot as plt
import cv2

def R7(image):
    """Filtro con Regla 7 para deteccion de fuego """
    im_read = cv2.imread(image)

    fil, col, ch = im_read.shape

    im_YCbCr = cv2.cvtColor(im_read, cv2.COLOR_RGB2YCrCb)
    im_filtro = np.zeros((fil, col))

    for idx in range(0, fil):
        for idy in range(0, col):
            if im_YCbCr[idx][idy][1] <= 120 and im_YCbCr[idx][idy][2] >= 150:
                im_filtro[idx][idy] = 1
            else:
                im_filtro[idx][idy] = 0

    cv2.imshow("Rule 7", im_filtro)
    cv2.waitKey(0)
    cv2.destroyAllWindows()